Makefile

```diff
@@ -282,8 +285,13 @@ build/src/libultra/libc/llcvt.o: src/libultra/libc/llcvt.c
 	@$(OBJDUMP) -d $@ > $(@:.o=.s)
 
 # Build C files from assets
+
+assets/%.c: assets/%.xml
+	$(ZAPD) bsf -eh -i $< -o $(dir $<)
+	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o build/$(@:.c=.o) $@
+
 build/%.inc.c: %.png
-	$(ZAPD) btex -eh -tt $(lastword ,$(subst ., ,$(basename $<))) -i $< -o $@
+	$(ZAPD) btex -eh -tt $(subst .,,$(suffix $*)) -i $< -o $@
```

Comment:
What is this doing?

Reply:
Presumably it does the exact same thing, extract the texture format from the filename. I updated it with the version used in OoT.